WifiART/src/userinterface/ZView.java

```diff
@@ -99,5 +102,101 @@ private void setup() {
         this.add(locationMovementPanelPanel);
     }
 
+    // Getters
+
+    /**
+     * Get the floor number.
+     *
+     * @return The floor number.
+     * @throws NumberFormatException If the user did not enter an integer.
+     */
+    public int getFloor() throws NumberFormatException {
+        return Integer.parseInt(floorField.getText());
+    }
+
+    /**
+     * Get the height above the floor.
+     *
+     * @return The height above the floor, in meters.
+     * @throws NumberFormatException If the user did not enter a decimal number.
+     */
+    public double getHeightAboveFloor() throws NumberFormatException {
+        return Double.parseDouble(heightAboveFloorField.getText());
+    }
+
+    /**
+     * Get the uncertainty for the height above the floor.
+     *
+     * @return The height above floor uncertainty, in meters.
+     * @throws NumberFormatException If the user did not enter a decimal number.
+     */
+    public double getHeightAboveFloorUncertainty() throws NumberFormatException {
+        return Double.parseDouble(heightAboveFloorUncertaintyField.getText());
+    }
+
+    /**
+     * Get the String descriptor for the STA location movement.
+     *
+     * @return The String descriptor.
+     */
+    public String getLocationMovement() {
+        if (fixedLocationMovementRadioButton.isSelected()) {
+            return fixedLocationMovementRadioButton.getText();
+        } else if (variableLocationMovementRadioButton.isSelected()) {
+            return variableLocationMovementRadioButton.getText();
+        } else if (unknownLocationMovementRadioButton.isSelected()) {
+            return unknownLocationMovementRadioButton.getText();
+        }
+        return null; // Only null if the user has not chosen an option yet.
```

Comment:
Could/Should we treat "user not choosing an option" as unknown?

Reply:
Technically the "return null" line will never be reached since this method is triggered by an JRadioButton being pressed. If this method ever returns null then there is a bug in the code. Do you think this would make more sense if I throw an exception here instead of returning null?

Reply:
I see, in that case maybe Precondition checks is more suitable. It throws RuntimeException when unexpected conditions (bugs instead of possible edge cases) occur.

Reply:
In this case, it isn't a matter of checking an input for some precondition. I have to manually check whether each button is selected before I can know if there is a bug. I think throwing a RuntimeException here is a good idea, though, rather than the caller of this method encountering a NullPointerException if there is a bug.